molgenis-api-data/src/main/java/org/molgenis/api/data/v2/RestControllerV2.java

```diff
@@ -142,29 +145,27 @@ public RestControllerV2(
       RestService restService,
       LocalizationService localizationService,
       PermissionSystemService permissionSystemService,
-      RepositoryCopier repoCopier) {
+      RepositoryCopier repoCopier,
+      VersionController versionController) {
     this.dataService = requireNonNull(dataService);
     this.permissionService = requireNonNull(permissionService);
     this.restService = requireNonNull(restService);
     this.localizationService = requireNonNull(localizationService);
     this.permissionSystemService = requireNonNull(permissionSystemService);
     this.repoCopier = requireNonNull(repoCopier);
+    this.versionController = requireNonNull(versionController);
   }
 
   @Autowired
+  @Deprecated
```

Comment:
I suggest that the only change to this controller is adding the @Deprecated annotation as well as a @deprecated javadoc annotation that refers to the preferred method.